### ISSUE TITLE
More compact toString for classes with many optional fields

### ISSUE THREAD

**Issue by rgabbard-bbn** [immutables/immutables#293]
We sometimes have classes with a large number of optional fields, most of which are usually absent.  For greater human-friendliness, it would be useful if there were a style which would alter the behavior of the generated `toString` methods so that absent fields are omitted and present optional fields are more compactly rendered.  For example, instead of 

```
Name{type=LOC, span=Amurka, score=Optional.of(1.0), transliteration=Optional.absent(), externalID=Optional.absent()}
```

you would have
```
Name{type=LOC, span=Amurka, score=1.0}
```

**Comment by elucash** [comment]
We actually had something like this at some point. But overall, after some though, I think that we might be in better position to implement what you're suggesting without a style flag, but as a normal behavior. The format of the toString should not be something people should rely on, so I would not be afraid to introduce the change. Thank you!

**Comment by rgabbard-bbn** [comment]
Even better, thanks!

**Comment by elucash** [comment]
Released in v 2.1.16 Thank you!
